From fc3c7c76409d00d760207eccc2e5f890ef90919b Mon Sep 17 00:00:00 2001
Message-Id: <fc3c7c76409d00d760207eccc2e5f890ef90919b.1303134271.git.jdenemar@redhat.com>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Thu, 14 Apr 2011 17:10:12 +0200
Subject: [PATCH] network: Fix NULL dereference during error recovery

This fixes: https://bugzilla.redhat.com/show_bug.cgi?id=696660

While starting a network, if brSetForwardDelay() fails, we go to err1
where we want to access macTapIfName variable which was just
VIR_FREE'd a few lines above. Instead, keep macTapIfName until we are
certain of success.
(cherry picked from commit 2444c411cabf4df583c4116dd730e8583eb19c69)

Conflict in src/network/bridge_driver.c. Resolved.
---
 src/network/bridge_driver.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/network/bridge_driver.c b/src/network/bridge_driver.c
index 20c3ddc..11671c9 100644
--- a/src/network/bridge_driver.c
+++ b/src/network/bridge_driver.c
@@ -1609,7 +1609,7 @@ networkStartNetworkDaemon(struct network_driver *driver,
     bool v4present = false, v6present = false;
     virErrorPtr save_err = NULL;
     virNetworkIpDefPtr ipdef;
-    char *macTapIfName;
+    char *macTapIfName = NULL;
 
     if (virNetworkObjIsActive(network)) {
         networkReportError(VIR_ERR_INTERNAL_ERROR,
@@ -1650,7 +1650,6 @@ networkStartNetworkDaemon(struct network_driver *driver,
             VIR_FREE(macTapIfName);
             goto err0;
         }
-        VIR_FREE(macTapIfName);
     }
 
     /* Set bridge options */
@@ -1724,6 +1723,8 @@ networkStartNetworkDaemon(struct network_driver *driver,
         goto err5;
     }
 
+    VIR_FREE(macTapIfName);
+    VIR_INFO(_("Starting up network '%s'"), network->def->name);
     network->active = 1;
 
     return 0;
@@ -1770,6 +1771,7 @@ networkStartNetworkDaemon(struct network_driver *driver,
                  macTapIfName, network->def->bridge,
                  virStrerror(err, ebuf, sizeof ebuf));
     }
+    VIR_FREE(macTapIfName);
 
  err0:
     if (!save_err)
-- 
1.7.5.rc1

